client/service/src/storage_monitor.rs

```diff
@@ -0,0 +1,158 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{config::Configuration, error::Error};
+use futures::StreamExt;
+use nix::{errno::Errno, sys::statvfs::statvfs};
+use notify::{Config, Event, RecommendedWatcher, RecursiveMode, Watcher};
+use sc_utils::mpsc::{tracing_unbounded, TracingUnboundedReceiver};
+use std::{
+	path::{Path, PathBuf},
+	time::{Duration, Instant},
+};
+
+const LOG_TARGET: &str = "storage-monitor";
+const THROTTLE_PERIOD: std::time::Duration = Duration::from_secs(2);
```

Comment:
also wondering how did you end up choosing 2 sec? why not 5 or 10s? or even 60s? do you know the avg disk space growth rate in polkadot?

Reply:
Just a guess. The syscall is cheap, should not affect the system. I did not measure the usage by node, but there also can be other consumers of storage running on the machine.